### ISSUE TITLE
no_grad and torch2trt

### ISSUE THREAD

**Issue by wico-silva** [NVIDIA-AI-IOT/torch2trt#727]
It doesn't seem to be documented anywhere what's the guideline regarding torch2trt and gradients. I assume gradients are never needed since the tests are all wrapped in `with torch.no_grad()`.

If it's never needed why not decorate the `torch2trt()` conversion function with `torch.no_grad()`? That way the user doesn't need to worry about whether or not torch2trt needs gradients and the user's codebase is not polluted with `with torch.no_grad()`.

If there are use-cases where it makes sense to enable gradients with torch2trt then they should be documented. Avoiding the risk that users let gradients enabled incurring huge GPU memory cost during conversion when they would probably not need it.
